### DISCUSSION TITLE
finished Event Listener Not Triggering When Specifying Animation Name in play() Method

### DISCUSSION THREAD

**Discussion by samsloan** [google/model-viewer#4525] [Q&A] [answered]
Hello everyone,

I am working with <model-viewer> and encountering an issue where the finished event listener does not trigger when I specify the animation name in the play() method. I am trying to log a message to the console when the animation finishes playing. The event listener works as expected when I don’t specify the animation name, but fails to trigger when I do.

Here's a simplified version of my code:

This works:

```
function playAnim() {
  
  const modelViewer = document.querySelector('model-viewer');
  
  // Play the animation once
  modelViewer.play({ repetitions: 1 });

  // When the animation finishes playing forward, log to console
  modelViewer.addEventListener("finished", (ev) => {
    console.log('animation finished');
  });
}
```

This does not work:
```
function playAnim() {

  const modelViewer = document.querySelector('model-viewer');
 
  // Play the animation once
  modelViewer.play('SpecificAnimation', { repetitions: 1 });

  // When the animation finishes playing forward, log to console
  modelViewer.addEventListener("finished", (ev) => {
    console.log('animation finished');
  });
}
```

In both cases, the animation plays successfully, but in the latter case, the finished event listener does not get triggered, and thus, 'animation finished' is not logged to the console.

I have also tried setting the attribute all different ways before playing the animation so I don't have to specify in the play method like:

```
modelViewer.setAttribute('animation-name', 'ArmatureAction');

 modelViewer.play({ repetitions: 1 });

```
But no success with this either

Any insights or suggestions would be greatly appreciated. Cheers!

Sam
[customViewer02_DEBUG.zip](https://github.com/google/model-viewer/files/13068151/customViewer02_DEBUG.zip)

**Comment by elalish** [comment, marked as answer]
First, I don't believe our `play` method takes an animation name as an input. Presumably this call is just failing? Second, when you set an attribute, make sure to `await modelViewer.updateComplete` - this is a method we inherit from Lit - before invoking `play`. I believe this is all just an async problem.

**Reply by samsloan** [reply]
await modelViewer.updateComplete

This is what I was missing. Very new to the world of async. My apologies for missing this!

Thank you
